addons/mrp/models/product.py

```diff
@@ -348,6 +349,16 @@ def action_compute_bom_days(self):
         warehouse = self.env['stock.warehouse'].search([('company_id', '=', company_id)], limit=1)
         for product in self:
             bom_data = self.env['report.mrp.report_bom_structure'].with_context(minimized=True)._get_bom_data(bom_by_products[product], warehouse, product, ignore_stock=True)
+            if bom_data.get('availability_state') == 'unavailable' and not bom_data.get('components_available', True):
```

Comment:
I would add a notification. it's a bit less intrusive, also you could potentially comes from a different action that the one you choose so can be confusing 
see : 
https://github.com/odoo/odoo/blob/bcc0e9b47bb2db8c2b0bc24ad2257057a6bc6ef2/addons/stock/wizard/product_replenish.py#L119-L126

Reply:
applied :)